frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/disks/validate-storage-profile.ts

```diff
@@ -0,0 +1,75 @@
+import { listPage } from '@console/cypress-integration-tests/views/list-page';
+import { TEMPLATE_NAME } from '../../const/index';
+import { ProvisionSource } from '../../enums/provisionSource';
+import { testName } from '../../support';
+import { VirtualMachineData } from '../../types/vm';
+import { virtualization } from '../../view/virtualization';
+import { vm } from '../../view/vm';
+
+const vmData: VirtualMachineData = {
+  name: `validate-storage-profile-${testName}`,
+  namespace: testName,
+  template: TEMPLATE_NAME,
+  provisionSource: ProvisionSource.REGISTRY,
+  pvcSize: '1',
+  sshEnable: false,
+  startOnCreation: false,
+};
+
+describe('ID(CNV-6923) Verify storageProfile with a fake storageClass', () => {
+  before(() => {
+    cy.Login();
+    cy.visit('/');
+    cy.createProject(testName);
+    cy.createStorageClass(); // ensure SC existence
+    cy.editStorageProfile(); // ensure SP has default properties for AccessMode and VolumeMode
+    vm.create(vmData);
+    virtualization.vms.visit();
+    cy.byLegacyTestID(vmData.name)
+      .should('exist')
+      .click();
+  });
+
+  after(() => {
+    cy.deleteResource({
+      kind: 'VirtualMachine',
+      metadata: {
+        name: vmData.name,
+        namespace: vmData.namespace,
+      },
+    });
+    cy.deleteStorageClass();
+    cy.deleteStorageProfile();
+    cy.deleteResource({
+      kind: 'Namespace',
+      metadata: {
+        name: testName,
+      },
+    });
+  });
+
+  it('ID(CNV-6922) Verify storageProfile in add disk modal', () => {
+    cy.byLegacyTestID('horizontal-link-Disks').click();
```

Comment:
AFAIU `byTestID` is `data-test=` and `byLegacyTestID` is `data-test-id=`

can you use this PR to refactor all the legacy test ids used in this test to be `data-test=`
(and check if they are used in other tests and refactor there also) ?
it's not fun to see `byTestID` and `byLegacyTestID` in the same test

Reply:
we will do it in a separate PR